### PR TITLE
[SU-267] Allow filtering workspaces by cloud provider

### PR DESCRIPTION
**Pull request by nawatts**
This adds an option to filter the workspaces list by cloud provider (GCP or Azure).

https://user-images.githubusercontent.com/1156625/211655489-c95510cb-3928-4101-a683-e60dd800511d.mov

### CODE REVIEW

**Review by marctalbott — APPROVED** [review]

File: src/libs/workspace-utils.ts [hunk 1]
```diff
@@ -4,6 +4,12 @@ export const cloudProviderTypes: Record<CloudProviderType, CloudProviderType> =
   AZURE: 'AZURE',
   GCP: 'GCP'
 }
+
+export const cloudProviderLabels: Record<CloudProviderType, string> = {
+  AZURE: 'Microsoft Azure',
```
Comment (cahrens) [inline src/libs/workspace-utils.ts]: These labels are all over the place. I would love to see a cleanup PR that has all the different locations reference these new constants.

https://github.com/DataBiosphere/terra-ui/search?q=%22Microsoft+Azure%22

At the least, could you change the 2 usages in the workspaces directory?

**Comment by cahrens** [comment]
I do wonder about adding this feature for everyone, right now. Only the alpha preview users (and us for testing purposes) will be creating Azure workspaces, so for the vast majority of folks the only choice is Google Cloud Platform. And for the preview users, I believe the thought is that they will ONLY have Azure workspaces… so they don't really need this either. I'm worried that it adds yet another control to a busy area, and it won't provide value. 

I have found myself wishing that the search option would respect Azure vs. GCP for this purpose.

File: src/libs/workspace-utils.ts (hunk 1 above)
Comment (nawatts) [reply src/libs/workspace-utils.ts]: I may open a separate PR for this since there's some discussion about when/if to merge the filter.

**Comment by nawatts** [comment]
I think the use case for this is mainly filtering public/featured workspaces. Since most users will only be using one cloud platform, they'll only be interested in the public/featured workspaces on that platform.

**Comment by cahrens** [comment]
> I think the use case for this is mainly filtering public/featured workspaces. Since most users will only be using one cloud platform, they'll only be interested in the public/featured workspaces on that platform.

Oh, I wasn't thinking of featured workspaces. Are we planning to show Azure featured workspaces to everyone? Or do you have to be in the preview group?

**Comment by nawatts** [comment]
My understanding was that we'd show all featured workspaces to everyone, but I don't know if that question has been asked. There is precedent (in login with Microsoft) for showing to everyone features that are only relevant to preview users.

**Comment by cahrens** [comment]
> My understanding was that we'd show all featured workspaces to everyone, but I don't know if that question has been asked. There is precedent (in login with Microsoft) for showing to everyone features that are only relevant to preview users.

Well, I guess as UX is on board with this, I won't hold it up! I personally am not a fan of this filtering UI though and wish we had a more concise UI for faceted search.

**Review by cahrens — APPROVED** [review]

**Comment by admull** [comment]
> My understanding was that we'd show all featured workspaces to everyone, but I don't know if that question has been asked. There is precedent (in login with Microsoft) for showing to everyone features that are only relevant to preview users.

Yes, I think this is the compromise, since Azure users can also see GCP featured workspaces. We want featured workspaces to be discoverable and advertised to incentivize people to join the cloud platform that has the data they want.

File: src/libs/workspace-utils.ts (hunk 1 above)
Comment (nawatts) [reply src/libs/workspace-utils.ts]: #3663 addresses some of this duplication.

**Comment by nawatts** [comment]
Rebased to resolve merge conflict.

**Comment by nawatts** [comment]
Changed "cloud provider" to "cloud platform" since that's the term already used in the workspaces list. Rebased to disable flaky tests.

**Comment by nawatts** [comment]
Reordered the filters (moved cloud platform to the end) to match the order of table columns / icons in the workspace cards.

Screen Shot 2023-01-20 at 1 39 39 PM

**Pull request merged**